doc/releases/changelog-0.36.0.md

```diff
@@ -303,68 +383,90 @@
+  new operator arithmetic in cases where they offer additional functionality not 
+  available using legacy operators.
   [(#5253)](https://github.com/PennyLaneAI/pennylane/pull/5253)
   [(#5410)](https://github.com/PennyLaneAI/pennylane/pull/5410)
   [(#5411)](https://github.com/PennyLaneAI/pennylane/pull/5411)
   [(#5421)](https://github.com/PennyLaneAI/pennylane/pull/5421)
 
-* A new `Prod.obs` property is introduced to smoothen the transition of the new operator arithmetic system.
-  In particular, this aims at preventing breaking code that uses `Tensor.obs`. This is immediately deprecated.
-  Moving forward, we recommend using `op.operands`.
+* `Prod` instances temporarily had a new `obs` property, which helped smoothen the 
+  transition of the new operator arithmetic system. In particular, this was aimed at preventing 
+  breaking code that uses `Tensor.obs`. This has been immediately deprecated. Moving 
+  forward, we recommend using `op.operands`.
   [(#5539)](https://github.com/PennyLaneAI/pennylane/pull/5539)
   
-* `ApproxTimeEvolution` is now compatible with any operator that defines a `pauli_rep`.
+* `qml.ApproxTimeEvolution` is now compatible with any operator that has a defined `pauli_rep`.
   [(#5362)](https://github.com/PennyLaneAI/pennylane/pull/5362)
 
-* `Hamiltonian.pauli_rep` is now defined if the hamiltonian is a linear combination of paulis.
+* `Hamiltonian.pauli_rep` is now defined if the hamiltonian is a linear combination of Pauli operators.
   [(#5377)](https://github.com/PennyLaneAI/pennylane/pull/5377)
 
-* `Prod.eigvals()` is now compatible with Qudit operators.
+* `Prod` instances created with qutrit operators now have a defined `eigvals()` method.
   [(#5400)](https://github.com/PennyLaneAI/pennylane/pull/5400)
 
-* `qml.transforms.hamiltonian_expand` can now handle multi-term observables with a constant offset.
+* `qml.transforms.hamiltonian_expand` can now handle multi-term observables with a constant offset (i.e., terms like `qml.I()`).
   [(#5414)](https://github.com/PennyLaneAI/pennylane/pull/5414)
 
-* `taper_operation` method is compatible with new operator arithmetic.
+* `qml.qchem.taper_operation` is now compatible with the new operator arithmetic.
   [(#5326)](https://github.com/PennyLaneAI/pennylane/pull/5326)
 
-* Removed the warning that an observable might not be hermitian in `qnode` executions. This enables jit-compilation.
+* The warning for an observable that might not be hermitian in QNode executions has been removed. This enables jit-compilation.
   [(#5506)](https://github.com/PennyLaneAI/pennylane/pull/5506)
 
 * `qml.transforms.split_non_commuting` will now work with single-term operator arithmetic.
   [(#5314)](https://github.com/PennyLaneAI/pennylane/pull/5314)
 
-* `LinearCombination` and `Sum` now accept `_grouping_indices` on initialization.
+* `LinearCombination` and `Sum` now accept `_grouping_indices` on initialization. This addition is relevant to developers only. 
```

Comment:
This ok? Feels like an entry that won't be relevant to users, so I left it kind of vague and added "this is for developers"